packages/editor/src/components/assets/ScenesPanel.tsx

```diff
@@ -138,7 +138,7 @@ export default function ScenesPanel() {
     setRenaming(false)
     const currentURL = loadedScene!.assetURL
     const newURL = currentURL.replace(currentURL.split('/').pop()!, newName + '.gltf')
-    const newData = await renameScene(id, newURL)
+    const newData = await renameScene(id, newURL, loadedScene!.projectName)
```

Comment:
While I agree with what you're saying @jonathan-casarrubias this is premature. We are doing this all over the place, and are focusing on architectural design rather than the smallest details, which is part of a longstanding effort to achieve what you are suggesting since the project began 4 years ago, trading off architectural design and immediate requirements over strict type safety.